### DISCUSSION TITLE
Plugin of Visual Studio is not working, permanently hangs

### DISCUSSION THREAD

**Discussion by cmp-nct** [community/community#38633] [Copilot Conversations]
I thought I was unlucky with the IntelliJ Plugin, though the Jetbrains plugin, as minimalistic as it is, works without issues.
I've tried a lot on Visual Studio 2022, the plugin permanently stops working in a variety of ways.
Most times you need to restart the whole IDE, sometimes it's enough to just close the tab and open it again.
Sometimes the plugin haunts you with code that does not exist anymore, as if internal buffers are not refreshed.

If I had started out with testing Copilot on VS I would without any doubt have uninstalled it within half an hour and never tried again.

**Comment by tilmann1** [comment]
@cmp-nct Can you please share more details of the issues that you are seeing? We'll try to reproduce them so that we can fix the underlying issues.

(1) Which version of Visual Studio are you using?
(2) Which version of the Copilot Visual Studio extension are you using?
(3) Can you please describe the series of steps that cause Visual Studio to hang?

Thanks in advance!

**Reply by cmp-nct** [reply]
I have a fresh installation of VS for a C# project, I used the official installer and after noticing serious problems with Copilot I updated to 17.4.0 which didn't make a change.
Copilot is on auto update, so that's 1.56.0.2 currently with the RPC agent at 1.56.7152 

I first used it natively Intellisense which caused a lot of issues with copilot, I switched to Resharper Trial and disabled the IDE and Copilot suggestions to compete with each other. So currently I have ReSharper active, it helped a bit on some issues.

I do not have a very complicated project, it's a few .cs files with 200-600 lines each.
Here are the worst problems I am having (though some of them seem to be reported since the closed beta)

In 9/10 cases I use copilot in the middle of existing code, so it has to add or modify functionality.
I've the impression the plugin was developed to generate new code only which is an edge case scenario for developers.

1) Copilot just stops suggesting when typing. 
This can happen literally anywhere, I noticed it happening mostly when I am at a "difficult" spot.
By difficult I mean that I delete a line 2-3 times because the solution/code did not work as expected so I need another code.
Suddenly copilot will just become unresponsive.

1.1) This sometimes can be resolved by manually triggering Inline completion, then it works.
But from there on I need to manually trigger it every time I need a suggestion, it stopped automatic ones.
It makes no difference if I use "wait for pause" or not. It acts as if I was on a rate-limiter for typing too fast, but I think that's not the case.
Often this does not work, copilot stays unresponsive.

1.2) Sometimes this can be resolved by saving the document, switching to another file/tab, triggering copilot there and going back.
However in 1/2 of the cases copilot will stop working in all tabs.

1.3) Often this can be resolved by CTRL+ALT+ENTER to open the side panel, it will "Load solutions" then you can use those solutions in the IDE again.
However, "Accept solution" often does not work. It doesn't trigger the IDE insert. You need to go to the IDE and manually cycle to the solution you wanted to accept.
From my experience copilot is in an unstable state from here on, it will continue to work for minutes up to 1-2 hours.
After this it will be "dead" and "loading solutions" will not stop anymore.
I tested that over night, copilot was still "loading solutions" when I came back after 7 hours.
After this happened you need to restart the IDE, nothing else works.
I even tried killing the RPC process but there seems to be no restart logic, so that doesn't help.

1.4) Sometimes this can be solved by saving, closing the tab and opening the file again.



2) Garbage code inserts
I guess you have a company directive to not fix that.
In maybe 8/10 cases copilot suggests one line of code that I need and 5-50 lines of code that are garbage and need to be carefully removed after accepting.
The plugin does not understand if a full function should be added (generating new code) or if it is just there to extend one line or a part of a line.
So it inserts code and often closing brackets (or closing comment tags) inside of existing code rendering it duplicate or dysfunctional.
Many times I discovered something like a left over "};" in the middle of my project, it can be difficult to find these errors if you miss them.

On Jetbrains I wrote a "hack" to fix that which inserts code word-by-word but frankly .. that should be part of the minimal functionality of the plugin.
